### ISSUE TITLE
reinvestigate missing line numbers in scala classes

### ISSUE THREAD

**Issue by fommil** [ensime/ensime-server#760]
I'm sure I introduced a regression in the classfile indexer... this info is definitely in there.

**Comment by aemoncannon** [comment]
I was under the impression that this information is optional, depending on
the debug level passed to scalac. Could that be the issue?

**Comment by fommil** [comment]
I am not 100% sure. I see line numbers in stack traces, so I'm inclined to believe they are there to some extent.

**Comment by edani** [comment]
Line numbers were there for the libraries I checked (and the ENSIME log confrms it in "Search:" log lines)

I fixed the main problem which was incorrect translation of a scala symbol into a JVM class name.  Here is the breakdown of remaining issues:

- Cosmetic: line numbers are often off by one or more. This is more visible for class/trait/object definitions since we use the constructor's line number, which doesn't match the actual class declaration
- for traits there is still something wrong, I don't think I fixed the naming
- polymorphic methods: we don't index methods by their parameter types, and we'd have to translate scala types into their JVM equivalent. So right now we index one of the methods and not the others.

So I think we'd get 90% there using the classfile indexer's lines, it was just easier to get to 100% through the pres. compiler.

Note that linking to .java sources is still a bit off because of the issues I described.
